.pylintrc

```diff
@@ -20,7 +20,8 @@ enable=
 disable=bad-continuation,
         missing-type-doc,
         missing-return-type-doc,
-        too-few-public-methods
+        too-few-public-methods,
+        missing-yield-type-doc
```

Comment:
I've no idea what this wants and I can't find any documentation on it.

Reply:
It wants you to add an `:rtype:` because you have a docstring that has a `:return:` but doesn't have an `:rtype:` (and PyLint calls it a "yield type" because said method happens to use `yield` not `return`, but it's still `:return:` and `:rtype:` in the docstring, there's no `:yield:` and `:ytype:` AFAIK).

We have already disabled return type docstrings warnings in LMS so I'm happy to disable it here as well.

* * *

Googling PyLint messages never works. Instead:

```
$ tox -qqe lint --run-command 'pylint --help-msg missing-yield-type-doc'
:missing-yield-type-doc (W9014): *Missing yield type documentation*
  Please document the type yielded by this method. This message belongs to the
  parameter_documentation checker.
```

The other method that turns up results is to search PyLint's source on GitHub: <https://github.com/PyCQA/pylint/search?q=missing-yield-type-doc>. That will often find you the PR that added the message: <https://github.com/PyCQA/pylint/pull/1008>

Reply:
Right ok. I knew there was some type of Google varient, but I couldn't find any detail on what you're supposed to do.